smarts/sstudio/generators.py

```diff
@@ -332,3 +333,63 @@ def _resolve_log_dir(self, log_dir):
             log_dir = make_dir_in_smarts_log_dir("_duarouter_routing")
 
         return os.path.abspath(log_dir)
+    
+
+def tripwritexml(resolve_route, traffic, doc):
```

Comment:
This tabbing seems wrong.

```suggestion
    def tripwritexml(resolve_route, traffic, doc):
```